sentry_streams/sentry_streams/examples/batching.py

```diff
@@ -1,24 +1,25 @@
 from sentry_kafka_schemas.schema_types.ingest_metrics_v1 import IngestMetric
 
-from sentry_streams.pipeline import Batch, FlatMap, streaming_source
-from sentry_streams.pipeline.batch import unbatch
-from sentry_streams.pipeline.chain import Parser, Serializer, StreamSink
+from sentry_streams.pipeline import Batch, streaming_source
+from sentry_streams.pipeline.chain import (
+    BatchParser,
+    ExtensibleChain,
+    Serializer,
+    StreamSink,
+)
+from sentry_streams.pipeline.message import Message
 
 pipeline = streaming_source(
     name="myinput",
     stream_name="ingest-metrics",
 )
 
 # TODO: Figure out why the concrete type of InputType is not showing up in the type hint of chain1
-parsed = pipeline.apply("parser", Parser(msg_type=IngestMetric))
-
-batched = parsed.apply("mybatch", Batch(batch_size=5))  # User simply provides the batch size
-
-unbatched = batched.apply(
-    "myunbatch",
-    FlatMap(function=unbatch),
-)
+parsed_batch: ExtensibleChain[Message[IngestMetric]] = pipeline.apply(
```

Comment:
Why do you need this type hint `parsed_batch: ExtensibleChain[Message[IngestMetric]]` ?